system/security/digisign/digisign.cpp

```diff
@@ -0,0 +1,247 @@
+            EVP_THROW("digiSign:EVP_DigestSignUpdate: %s");
+        }
+
+        //compute length of signature
+        size_t encMsgLen;
+        if (EVP_DigestSignFinal(RSASignCtx, nullptr, &encMsgLen) <= 0)
+        {
+            EVP_CLEANUP(priKey, RSASignCtx);
+            EVP_THROW("digiSign:EVP_DigestSignFinal1: %s");
+        }
+
+        if (encMsgLen == 0)
+        {
+            EVP_CLEANUP(priKey, RSASignCtx);
+            EVP_THROW("digiSign:EVP_DigestSignFinal length returned 0: %s");
+        }
+
+        //compute signature (signed digest)
+        unsigned char * encMsg = (unsigned char*) malloc(encMsgLen);
+        if (encMsg == nullptr)
+        {
+            EVP_CLEANUP(priKey, RSASignCtx);
+            throw MakeStringException(-1, "digiSign:malloc(%ld) returned NULL",encMsgLen);
+        }
+
+        if (EVP_DigestSignFinal(RSASignCtx, encMsg, &encMsgLen) <= 0)
+        {
+            free(encMsg);
+            EVP_CLEANUP(priKey, RSASignCtx);
+            EVP_THROW("digiSign:EVP_DigestSignFinal2: %s");
+        }
+
+        //convert to base64
+        JBASE64_Encode(encMsg, encMsgLen, b64Signature, false);
+
+        //cleanup
+        free(encMsg);
+        EVP_CLEANUP(priKey, RSASignCtx);
+        return true;//success
+#else
+        throw MakeStringException(-1, "digiSign:Platform built without OPENSSL");
+        return false;
+#endif
+    }
+
+
+    //Verify the given text was used to create the given digital signature
+    bool digiVerify(const char * text, StringBuffer & b64Signature)
+    {
+        if (!verifyingConfigured)
+            throw MakeStringException(-1, "digiVerify:Verifying Digital Signatures not configured");
+
+        CriticalBlock b(digiVerifyCrit);
+
+        if (publicKeyBuff.isEmpty())
+            publicKeyBuff.loadFile(publicKeyFile.str());
+        if (publicKeyBuff.isEmpty())
+            throw MakeStringException(-1, "digiSign:Cannot load public key file");
+
+#ifdef _USE_OPENSSL
+        BIO * keybio = BIO_new_mem_buf((void*) publicKeyBuff.str(), -1);
+        if (keybio == nullptr)
+        {
+            EVP_THROW("digiVerify:BIO_new_mem_buf: %s");
+        }
+        RSA * rsa = PEM_read_bio_RSA_PUBKEY(keybio, nullptr, nullptr, nullptr);
+
+        EVP_PKEY* pubKey = EVP_PKEY_new();
+        EVP_PKEY_assign_RSA(pubKey, rsa);
+        EVP_MD_CTX * RSAVerifyCtx = EVP_MD_CTX_create();//allocate, initializes and return a digest context
+        if (RSAVerifyCtx == nullptr)
+        {
+            EVP_PKEY_free(pubKey);
+            BIO_free_all(keybio);
+            EVP_THROW("digiVerify:EVP_MD_CTX_create: %s");
+        }
+
+        if (EVP_DigestVerifyInit(RSAVerifyCtx, nullptr, EVP_sha256(), nullptr, pubKey) <= 0)
+        {
+            EVP_CLEANUP(pubKey, RSAVerifyCtx);//cleans allocated key and digest context
+            BIO_free_all(keybio);
+            EVP_THROW("digiVerify:EVP_DigestVerifyInit: %s");
+        }
+        BIO_free_all(keybio);
```

Comment:
minor: this is done earlier in the signing function. (avoiding the need for it to be duplicated)